src/main/java/com/main/service/impl/PersonServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.main.service.impl;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.main.dto.ResponseDto;
+import com.main.entity.Message;
+import com.main.entity.Person;
+import com.main.repository.PersonRepository;
+import com.main.service.PersonService;
+
+@Service
+public class PersonServiceImpl implements PersonService {
+
+	private PersonRepository personRepository;
+
+	public PersonServiceImpl(PersonRepository personRepository) {
+		this.personRepository = personRepository;
+	}
+
+	public ResponseDto<String> findAllMessagesByPersonId(Long personId, String word) {
+
+		ResponseDto<String> response = new ResponseDto<>();
+
+		// First, find the person by their ID
+		Person person = personRepository.findById(personId).orElse(null);
+
+		// If the person is found, retrieve all messages from that person
+		if (person != null) {
+			List<Message> result = personRepository.findAllMessagesByPersonIdAndSearchWord(personId, word);
+			if (!result.isEmpty()) {
+				response.setStatus(200);
+				response.setMessage("Data retrieved successfully");
+				response.setData(result.stream().map(Message::getContent).collect(Collectors.toList()));
+				return response;
+			} else {
+				response.setStatus(200);
+				response.setMessage("No Data found");
+				response.setData(Collections.emptyList());
+				return response;
+			}
+		}
+		response.setStatus(200);
```

Comment:
if user not found, it should return 404.
Previously I thought it was listing "Persons" since the Controller is called "PersonController" instead of "MessagesController". In this case it should be 404 if the user_id is not found.